### DISCUSSION TITLE
From dense adjacency matrix to edge_index

### DISCUSSION THREAD

**Discussion by AndreaPi** [pyg-team/pytorch_geometric#5144] [Q&A] [answered]
I have the following dense adjacency matrix:

```
adjacency_matrix = torch.tensor([[0., 1., 1., 1.],
                                 [1., 0., 1., 1.],
                                 [1., 1., 0., 1.],
                                 [1., 1., 1., 0.]])
```

This corresponds to a 4-nodes graph, where each node is connected each other node but not to itself.
![graph](https://user-images.githubusercontent.com/8233615/183052632-6a18dc7c-f809-49ff-a81d-fa87e921c68b.jpg)

 Usually I see that conversion from this matrix  to edge_index is done this way:

```
edge_index = (adjacency_matrix > 0).nonzero().t()
```

Why do we need `(adjacency_matrix > 0)`? Wouldn't we get exactly the same result with `adjacency_matrix.nonzero().t()`? This is the case for all examples I tried so far.

**Comment by Padarn** [comment, marked as answer]
I think these are equivalent unless your `adjacency_matrix` contains edge weights too

**Reply by rusty1s** [reply]
That is correct. Transferring the adjacency matrix to a mask first will result in unnecessary computation and memory overheads though.

**Reply by AndreaPi** [reply]
Hi @rusty1s,

with

> Transferring the adjacency matrix to a mask first 

you mean this operation `(adjacency_matrix > 0)`, right?

**Reply by rusty1s** [reply]
Yes, this will create a bool matrix of shape `[num_nodes, num_nodes]`.
